### PR TITLE
Add UISwitch on's property

### PR DESCRIPTION
**Pull request by dmcrodrigues**

### CODE REVIEW

File: Source/UIKit/UISwitch.swift [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    /// Wraps a switch's `on` state in a bindable property.
+    public var rex_on: MutableProperty<Bool> {
+
+        let property = associatedProperty(self, key: &onKey, initial: { $0.on }, setter: { $0.on = $1 })
+
+        property <~ rex_controlEvents(.ValueChanged)
+            .map { $0 as? UISwitch }
+            .ignoreNil()
+            .map { $0.on }
```
Comment (RuiAAPeres) [inline Source/UIKit/UISwitch.swift]: Instead of: 

```
.map { $0 as? UISwitch }
.ignoreNil()
```

You can go with:

```
.filterMap { $0 as? UISwitch }
```

Comment (dmcrodrigues) [inline Source/UIKit/UISwitch.swift]: Wow, that's pretty neat :+1:

File: Source/UIKit/UISwitch.swift [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+extension UISwitch {
+
+    /// Wraps a switch's `on` state in a bindable property.
+    public var rex_on: MutableProperty<Bool> {
+
+        let property = associatedProperty(self, key: &onKey, initial: { $0.on }, setter: { $0.on = $1 })
+
+        property <~ rex_controlEvents(.ValueChanged)
+            .filterMap { $0 as? UISwitch }
+            .map { $0.on }
```
Comment (neilpa) [inline Source/UIKit/UISwitch.swift]: I think you can take this one step further and put it all in the `filterMap`

```
.filterMap { ($0 as? UISwitch)?.on }
```

I like to minimize the number of operators where possible.

File: Rex.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -8,6 +8,8 @@
 
 /* Begin PBXBuildFile section */
 		4238D5961B4D5950008534C0 /* NSTextField.swift in Sources */ = {isa = PBXBuildFile; fileRef = 4238D5951B4D5950008534C0 /* NSTextField.swift */; };
+		7D2AA99B1CB6EFEB008AB5C9 /* UISwitch.swift in Sources */ = {isa = PBXBuildFile; fileRef = 7D2AA99A1CB6EFEB008AB5C9 /* UISwitch.swift */; };
+		7D2AA99D1CB6F275008AB5C9 /* UISwitchTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 7D2AA99C1CB6F275008AB5C9 /* UISwitchTests.swift */; };
```
Comment (neilpa) [inline Rex.xcodeproj/project.pbxproj:12]: It's hard to decipher the xcproj from a diff. Did you also add this to the tvOS target?

File: Source/UIKit/UISwitch.swift (hunk 2 above)
Comment (dmcrodrigues) [inline Source/UIKit/UISwitch.swift]: I'll unify it with the `filterMap` like you suggested. I personally prefer having this kind of separation but makes sense minimize.

File: Rex.xcodeproj/project.pbxproj (hunk 3 above)
Comment (dmcrodrigues) [inline Rex.xcodeproj/project.pbxproj:12]: Is it available on tvOS? I wasn't aware of that.

File: Source/UIKit/UISwitch.swift (hunk 2 above)
Comment (neilpa) [inline Source/UIKit/UISwitch.swift]: Chained `*map` operators do look a bit cleaner and make sense semantically. However, signal transformations do have a real cost and can add up perf wise so I like to collapse them. In an ideal world there would be a way that happened automatically. But that's a personal pipe dream.

Comment (dmcrodrigues) [inline Source/UIKit/UISwitch.swift]: Good point, it's definitely something to consider! And that dream, I hope it becomes reality one day :sparkles:

File: Rex.xcodeproj/project.pbxproj (hunk 3 above)
Comment (dmcrodrigues) [inline Rex.xcodeproj/project.pbxproj:12]: @neilpa I remembered that I've checked this, adding it to the tvOS target we can see that's not available.

Comment (neilpa) [inline Rex.xcodeproj/project.pbxproj:12]: Doesn't look like it
https://developer.apple.com/library/tvos/documentation/UIKit/Reference/UIKit_Framework/index.html#classes

**Pull request merged**